include/pistache/os.h

```diff
@@ -139,4 +139,13 @@ class NotifyFd {
     int event_fd;
 };
 
+class FdScopeGuard {
+public:
+    FdScopeGuard(int &fd);
+
+    ~FdScopeGuard();
```

Comment:
Reference should not be used I think

Reply:
Actually I don't think that this class does anything that could be done in `FileBuffer`

Reply:
I agree.  ISTM that a much simpler solution is to ensure that the constructor for FileBuffer forces "fd_" to -1, and the destructor closes fd_ if it is >= 0.